content/blog/2016-07-23-pull-does-not-scale-or-does-it.md

```diff
@@ -0,0 +1,175 @@
+metrics sources.
+
+On a practical front, Prometheus makes it easy to configure the desired state
+of the world with its built-in support for a wide variety of service discovery
+mechanisms for cloud providers and container-scheduling systems: Consul,
+Marathon, Kubernetes, EC2, DNS-based SD, Azure, Zookeeper Serversets, and more.
+Prometheus also allows you to plug in your own custom mechanism if needed.
+
+## Accidentally DDoS-ing your monitoring
+
+Whether you use a pull-based or a push-based approach, any time-series database
+will fall over if you send it more samples than it can handle. However, we have
+made the experience that it's slightly more likely for a push-based approach to
+accidentally bring down your monitoring. If the control over what metrics get
+ingested from which instances is not centralized (in your monitoring system),
+then you run into the danger of experimental or rogue jobs suddenly pushing
+lots of garbage data into your production monitoring and bringing it down.
+There are still plenty of ways how this can happen with a pull-based approach
+(which only controls where to pull metrics from, but not the size and nature of
+the metrics payloads), but the risk is lower and such incidents can be
+fixed at a central point.
+
+## Real-world proof
+
+Besides the fact that Prometheus is already being used to monitor very large
+setups in the real world (like using it to [monitor millions of machines at
+DigitalOcean](http://promcon.io/talks/scaling_to_a_million_machines_with_prometheus/)),
+there are other prominent examples of pull-based monitoring being used
+successfully in the largest possible environments. Prometheus was inspired by
+Google's Borgmon, which was (and partially still is) used within Google to
+monitor all its critical production services using a pull-based approach. Any
+scaling issues we encountered with Borgmon at Google were not due its pull
+approach either. If a pull-based approach scales to a global environment with
+many tens of datacenters and hundreds of thousands of machines, you can hardly say
+that pull doesn't scale.
+
+## But there are other problems with pull!
+
+There are indeed setups that are hard to monitor with a pull-based approach.
+A prominent example is when you have many endpoints scattered around the
+world which are not directly reachable due to firewalls or complicated
+networking setups, and where it's infeasible to run a Prometheus server
+directly in each of the network segments. This is not quite the environment for
+which Prometheus was built, although workarounds are often possible ([via the
+Pushgateway or restructuring your setup](/docs/practices/pushing/)). In any
+case, these remaining concerns about pull-based monitoring are usually not
+scaling-related, but concern some other operational difficulty that pulling
+poses in certain environments.
+
+## All good then?
+
+We know that the debate about pulling versus pushing is a religious one and as
```

Comment:
A "religious" debate will just become more "religious" if you call them out as such. After having nicely demonstrated where pull vs. push matters and where not, perhaps it would have a more positive tone if you express the hope that the debate will in future be more about the aspects that matter and not about the red herring of scalability anymore?